core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2639,6 +2650,458 @@ public int getTenantId() throws UserStoreException {
+        if (returnedAttributes.contains(realmConfig.getUserStoreProperty(LDAPConstants.MEMBERSHIP_ATTRIBUTE))) {
+            DirContext dirContext = this.connectionSource.getContext();
+            String userNameProperty = realmConfig.getUserStoreProperty(LDAPConstants.USER_NAME_ATTRIBUTE);
+            String displayNameAttribute = realmConfig
+                    .getUserStoreProperty(LDAPConstants.DISPLAY_NAME_ATTRIBUTE);
+            String[] requiredAttributes = {userNameProperty, displayNameAttribute};
+
+            for (String user : userListFromSearch) {
+                try {
+                    Attributes userAttributes = dirContext.getAttributes(new CompositeName().add(user),
+                            requiredAttributes);
+
+                    String displayName = null;
+                    String userName = null;
+                    if (userAttributes != null) {
+                        Attribute userNameAttribute = userAttributes.get(userNameProperty);
+                        if (userNameAttribute != null) {
+                            userName = (String) userNameAttribute.get();
+                            if (log.isDebugEnabled()) {
+                                log.debug("UserName: " + userName);
+                            }
+                        }
+                        if (StringUtils.isNotEmpty(displayNameAttribute)) {
+                            Attribute displayAttribute = userAttributes.get(displayNameAttribute);
+                            if (displayAttribute != null) {
+                                displayName = (String) displayAttribute.get();
+                            }
+                        }
+                    }
+                    String domainName =
+                            realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+                    // Username will be null in the special case where the username attribute has changed to another
+                    // and having different userNameProperty than the current user-mgt.xml
+                    if (userName != null) {
+                        user = UserCoreUtil.getCombinedName(domainName, userName, displayName);
+                        users.add(user);
+                        if (log.isDebugEnabled()) {
+                            log.debug(user + " is added to the result list");
+                        }
+                    } else {
+                        // Skip listing users which are not applicable to current user-mgt.xml
+                        if (log.isDebugEnabled()) {
+                            log.debug("User " + user + " doesn't have the user name property : " +
+                                    userNameProperty);
+                        }
+                    }
+                } catch (NamingException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Error in reading user information in the user store for the user " +
```

Comment:
If debug is not enabled then user can't find any errors right?. Isn't it better to add an error log here?

Reply:
fixed